fcrepo-kernel-impl/src/test/java/org/fcrepo/kernel/impl/services/CreateResourceServiceImplTest.java

```diff
@@ -217,7 +217,7 @@ public void testNoParentRdf() throws Exception {
     public void testParentAg() {
         final FedoraId fedoraId = FedoraId.create(UUID.randomUUID().toString());
         final FedoraId childId = fedoraId.resolve("child");
-        containmentIndex.addContainedBy(transaction, fedoraId, childId);
+        containmentIndex.addContainedBy(transaction, FedoraId.getRepositoryRootId(), fedoraId);
```

Comment:
The original test was wrong because it was adding the containment relationship for the child before the child was created when it should have been adding the relationship for its parent. The child containment relationship is created when the resource is created in the test.

The test used to pass because the records were merged into the db using an upsert. However, they are now simply being inserted, which exposed the bug in the test.